consensus/bor/rest.go

```diff
@@ -96,13 +100,23 @@ func (h *HeimdallClient) FetchWithRetry(rawPath string, rawQuery string) (*Respo
 	u.Path = rawPath
 	u.RawQuery = rawQuery
 
+	// prepare a notify context to handle interrups
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
```

Comment:
We should not be relying on `os.Interrupt` at the low level of the code flow. Instead, we should use `Close()` functions that propagate when the low-level code has to stop. Note that there is already a `Close()` function as part of the consensus interface, so we have already an entry point to notify `HeimdallClient` that it has to stop.

An extra benefit of this is that you can test the `HeimdallClient` implementation without having to call `os.Interrupt` which I am not sure it is possible with the test framework of Go.

Usually, I approach these problems with a `Close()` function and a `closeCh` channel.

```
type HeimdallClient struct {
   closeCh chan struct{}
}

...
  &HeimdallClient{
    closeCh: make(chan struct{}),
  }
...

func (h *HeimdallClient) Close() {
   close(h.closeCh)
}

func (h *HeimdallClient) FetchWithRetry() {
  ... 
  for {
     select {
        case <-h.closeCh:
            return
        case <-time.After(5*time.Second):
     }
   }
}
```